### PR TITLE
RavenDB-8795

### PR DESCRIPTION
**Pull request by karmeli87**
- Resolve conflicts to a specific node was removed.
-  Conflicts are kept as revisions after they were resolved.

### CODE REVIEW

File: src/Raven.Server/Documents/ConflictsStorage.cs [hunk 1]
```diff
@@ -318,21 +319,32 @@ public bool HasHigherChangeVector(DocumentsOperationContext context, Slice prefi
...
                 {
-                    var etag = TableValueToEtag((int)ConflictsTable.Etag, ref conflictDocument.Reader);
-                    _documentsStorage.EnsureLastEtagIsPersisted(context, etag);
-
-                    var conflictChangeVector = TableValueToChangeVector(context, (int)ConflictsTable.ChangeVector, ref conflictDocument.Reader);
-                    changeVectors.Add(conflictChangeVector);
+                    var conflicted = TableValueToConflictDocument(context, ref conflictDocument.Reader);
+                    var collection = _documentsStorage.ExtractCollectionName(context, conflicted.Collection);
+
+                    if (conflicted.Doc != null)
```
Comment (ayende) [inline src/Raven.Server/Documents/ConflictsStorage.cs:325]: Braces here

File: src/Raven.Server/Documents/ConflictsStorage.cs [hunk 2]
```diff
@@ -318,21 +319,32 @@ public bool HasHigherChangeVector(DocumentsOperationContext context, Slice prefi
...
+                        {
+                            _documentsStorage.RevisionsStorage.DeleteRevision(context, key, conflicted.Collection, conflicted.ChangeVector);
+                        }
+                    }
+                    _documentsStorage.EnsureLastEtagIsPersisted(context, conflicted.Etag);
+                    changeVectors.Add(conflicted.ChangeVector);
 
-                    var flags = TableValueToFlags((int)ConflictsTable.Flags, ref conflictDocument.Reader);
-                    if ((flags & DocumentFlags.HasAttachments) != DocumentFlags.HasAttachments)
+                    if (conflicted.Flags.HasFlag(DocumentFlags.HasAttachments) == false)
```
Comment (ayende) [inline src/Raven.Server/Documents/ConflictsStorage.cs:340]: This does an allocation that the previous code didn't do
https://stackoverflow.com/questions/11665279/why-enums-hasflag-method-need-boxing

File: src/Raven.Server/Documents/Revisions/RevisionsStorage.cs [hunk 3]
```diff
@@ -183,17 +185,26 @@ public void InitializeFromDatabaseRecord(DatabaseRecord dbRecord)
...
-                return _emptyConfiguration;
+                return ConflictConfiguration.Default;
 
             if (Configuration.Collections != null &&
                 Configuration.Collections.TryGetValue(collection, out RevisionsCollectionConfiguration configuration))
             {
                 return configuration;
             }
-
+            if (flags.HasFlag(DocumentFlags.Resolved) || flags.HasFlag(DocumentFlags.Conflicted))
```
Comment (ayende) [inline src/Raven.Server/Documents/Revisions/RevisionsStorage.cs:204]: This allocates twice.

File: test/SlowTests/Server/Replication/ReplicationResolveConflictsOnConfigurationChange.cs [hunk 4]
```diff
@@ -77,56 +77,22 @@ public class ReplicationResolveConflictsOnConfigurationChange : ReplicationTestB
 
                 await SetReplicationConflictResolutionAsync(store1, StraightforwardConflictResolution.ResolveToLatest);
 
+                WaitForUserToContinueTheTest(store1);
```
Comment (ayende) [inline test/SlowTests/Server/Replication/ReplicationResolveConflictsOnConfigurationChange.cs:80]: Probably need to remove this?

**Review by ayende — CHANGES_REQUESTED** [review]

File: test/FastTests/Server/Documents/Revisions/RevisionsTests.cs [hunk 5]
```diff
@@ -92,19 +92,6 @@ public class RevisionsTests : RavenTestBase
         }
 
         [Fact]
-        public async Task GetRevisionsOfNotExistKey_WithRevisionsDisabled()
```
Comment (fitzchak) [inline test/FastTests/Server/Documents/Revisions/RevisionsTests.cs:95]: We removing the test? Better to change the assertion.

File: test/SlowTests/Server/Replication/ReplicationResolveToLeader.cs [hunk 6]
```diff
@@ -1,259 +0,0 @@
...
-                        resovlerDbId: documentDatabase.DbBase64Id);
-                await SetupReplicationAsync(store2, store1);
-                
-                Assert.True(WaitForDocument<User>(store1, "foo/bar", u => u.Name == "Oren"));
-                Assert.True(WaitForDocument<User>(store2, "foo/bar", u => u.Name == "Oren"));
-            }
-        }
-
-        [Fact]
-        public async Task ResolveToDatabaseComplex()
```
Comment (fitzchak) [inline test/SlowTests/Server/Replication/ReplicationResolveToLeader.cs:46]: Would it make sense to leave the tests here and change the assertion?

**Pull request merged**